### PR TITLE
BAEL-6465: How to handle NoSuchElementException when reading a file through a Scanner ?

### PR DESCRIPTION
**Pull request by azhwani**
Simple PR

### CODE REVIEW

File: core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        final Scanner scanner = new Scanner(new File(file));
+
+        return scanner.nextLine();
+    }
+
+    public static String readFileV2(String file) throws FileNotFoundException {
+        final Scanner scanner = new Scanner(new File(file));
+
+        return scanner.hasNextLine() ? scanner.nextLine() : "";
+    }
```
Comment (JavierPeris) [inline core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java:19]: We can improve this code using try-with-resources.

File: core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.baeldung.scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class ScannerNoSuchElementException {
+
+    public static String readFileV1(String file) throws FileNotFoundException {
```
Comment (JavierPeris) [inline core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java:9]: We can rename the parameter to path or pathname which I think is more meaningful.

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/scanner/ScannerNoSuchElementExceptionUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    void givenEmptyFilewhenReadFileWithScanner_thenThrowException() throws IOException {
+        Exception exception = assertThrows(NoSuchElementException.class, () -> {
+            ScannerNoSuchElementException.readFileV1("src/test/resources/emptyFile.txt");
+        });
+
+        assertEquals("No line found", exception.getMessage());
+    }
+
+    @Test
+    void givenEmptyFilewhenReadFileWithScanner_thenSuccess() throws IOException {
```
[the remark this answers is not in the document]
Comment (azhwani) [reply core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/scanner/ScannerNoSuchElementExceptionUnitTest.java:23]: Good catch ;) done

File: core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java (hunk 1 above)
Comment (azhwani) [reply core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java]: done, I replaced also the legacy class File with Path

File: core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java (hunk 2 above)
Comment (azhwani) [reply core-java-modules/core-java-io-apis-2/src/main/java/com/baeldung/scanner/ScannerNoSuchElementException.java]: done

File: core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/scanner/ScannerNoSuchElementExceptionUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+class ScannerNoSuchElementExceptionUnitTest {
+
+    @Test
+    void givenEmptyFile_whenUsingReadFileV1_thenThrowException() throws IOException {
+        Exception exception = assertThrows(NoSuchElementException.class, () -> {
+            ScannerNoSuchElementException.readFileV1("src/test/resources/emptyFile.txt");
+        });
+
+        assertEquals("No line found", exception.getMessage());
```
Comment (JavierPeris) [inline core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/scanner/ScannerNoSuchElementExceptionUnitTest.java:19]: The assert catch the possible exception so we can simpliy this method as:

```
void givenEmptyFile_whenUsingReadFileV1_thenThrowException() {
        Exception exception = assertThrows(NoSuchElementException.class, () ->
            ScannerNoSuchElementException.readFileV1("src/test/resources/emptyFile.txt"));

        assertEquals("No line found", exception.getMessage());
    }
```

**Pull request merged**